src/components/character-card/ui/CharacterCard.tsx

```diff
@@ -0,0 +1,63 @@
+import {
+  ICharacter,
+  useGetCharacterByIdQuery,
+} from '@/App/store/characters/api';
+import { fetchCharacterDetails } from '@/App/store/detailsSlice/detailsSlice';
+import { toggleSelection } from '@/App/store/selectedSlice/selectedSlice';
+import { RootState, useAppDispatch, useAppSelector } from '@/App/store/store';
+import { FC } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+interface ICharacterCardProps {
+  character: ICharacter;
+}
+
+export const CharacterCard: FC<ICharacterCardProps> = ({ character }) => {
+  const dispatch = useAppDispatch();
+  const selectedCharacters = useAppSelector(
+    (state: RootState) => state.selected.selected
+  );
+  const selectedDetails = useAppSelector(
+    (state: RootState) => state.details.character
+  );
+  const navigate = useNavigate();
+
+  const isSelected = selectedCharacters[character.id as string] ?? false;
```

Comment:
The name of the variable points that it should have type `boolean`, but it has `ICharacter | boolean`. 
```suggestion
  const isSelected = !!selectedCharacters[character.id as string];
```